### PR TITLE
specs: two reserved bytes

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/883

### CODE REVIEW

File: specs/src/specs/data_structures.md
Comment (rootulp) [inline specs/src/specs/data_structures.md:501]: [for reviewers] it is possible to confirm that an unsigned varint is used for the reserved byte by referencing https://github.com/celestiaorg/celestia-app/blob/9769b7508909a3ce8895b910b5236a8a94672621/pkg/shares/reserved_bytes.go#L11-L22

File: specs/src/specs/data_structures.md
Comment (SweeXordious) [inline specs/src/specs/data_structures.md:505]: [Non Blocking]
Doesn't it make sense not to use `<br>` as throughout the repo we're opting for new line?
I'm fine with switching to `<br>` notation

File: specs/src/specs/data_structures.md
Comment (SweeXordious) [inline specs/src/specs/data_structures.md:513]:
```suggestion
where reserved bytes would be `80` encoded as an unsigned varint (`[0b1010000, 0b0]`).
```

**Review by SweeXordious — COMMENTED** [review]
LGTM, left non blocking comments.
Will defer to others for approval

File: specs/src/specs/data_structures.md
Comment (rootulp) [reply specs/src/specs/data_structures.md:505]: the `<br>` is used to add an explicit newline between this sentence and the image. Without `<br>` the image may render on the same line as the sentence.

Screen Shot 2022-11-01 at 3 18 21 PM

Comment (SweeXordious) [reply specs/src/specs/data_structures.md:505]: Aaah got you, thanks :+1:

**Review by SweeXordious — COMMENTED** [review]
LGTM. Deferring to other reviewers

**Review by MSevey — DISMISSED** [review]
PR looks fine, and based on the Issue I assume the spec changes were all previously validated.

**Comment by rootulp** [comment]
Thanks for reviewing Matt! Will merge by end of next week to give @adlerjohn time to review.

File: specs/src/specs/data_structures.md
Comment (adlerjohn) [reply specs/src/specs/data_structures.md:505]: Can't you just add a newline?

File: specs/src/specs/data_structures.md
Comment (adlerjohn) [inline specs/src/specs/data_structures.md:508]: These are weird because you're truncating leading zeroes. Would prefer making these all 8 bits. Also, are you sure this is correct? With varint, if it takes more than one byte then the MSB is one.

Comment (rootulp) [reply specs/src/specs/data_structures.md:508]: > Would prefer making these all 8 bits

Agreed, will fix in an upcoming commit.

> Also, are you sure this is correct? With varint, if it takes more than one byte then the MSB is one.

Yes because `14` doesn't occupy more than one byte so the most significant bit is off for the first byte and `0` for the second byte. Ref: https://go.dev/play/p/TqP2QrcYh6z

Screenshot 2022-11-13 at 9 49 21 PM

File: specs/src/specs/data_structures.md
Comment (rootulp) [reply specs/src/specs/data_structures.md:505]: Yea, not sure why that didn't occur to me. Will fix in an upcoming commit.

File: specs/src/specs/data_structures.md
Comment (rootulp) [reply specs/src/specs/data_structures.md]: Resolved by https://github.com/celestiaorg/celestia-app/pull/939/commits/fae99b678722ee8ddf15a36a4766c1b64dee6f25

File: specs/src/specs/data_structures.md
Comment (rootulp) [reply specs/src/specs/data_structures.md]: Resolved by https://github.com/celestiaorg/celestia-app/pull/939/commits/ea4cea5ba03dfc27e0e74c33675b0725832a20b5

**Review by evan-forbes — APPROVED** [review]
:+1: LGTM

**Review by liamsi — APPROVED** [review]

**Comment by rootulp** [comment]
Will merge tomorrow if @adlerjohn has no more feedback

**Review by adlerjohn — APPROVED** [review]

**Pull request merged**
